test/main/assert/step.js

```diff
@@ -0,0 +1,44 @@
+QUnit.module( "assert.step" );
+
+QUnit.test( "pushes a failing assertion if no message is given", function( assert ) {
+	assert.expect( 2 );
+
+	var originalPushResult = assert.pushResult;
+	assert.pushResult = function pushResultStub( resultInfo ) {
+		assert.pushResult = originalPushResult;
+
+		assert.notOk( resultInfo.result );
+		assert.equal( resultInfo.message, "You must provide a message to assert.step" );
+	};
+
+	assert.step();
```

Comment:
I think it's more important to cover empty-string input, but we should probably include both in the test suite.